### ISSUE TITLE
Trainable variable defined inside loss function is not getting trained in keras.

### ISSUE THREAD

**Issue by codeloverdk** [keras-team/keras#12503]
I defined some tensorflow variable inside e custom loss function by setting trainable=true but while the training is going the values of the variables are not changing but i have non zero loss.

**Comment by ymodak** [comment]
Please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
